src/library_wasmfs_opfs.js

```diff
@@ -188,7 +188,15 @@ mergeInto(LibraryManager.library, {
   _wasmfs_opfs_insert_directory:
       async function(ctx, parent, namePtr, childIDPtr) {
     let name = UTF8ToString(namePtr);
-    let childID = await wasmfsOPFSGetOrCreateDir(parent, name, true);
+    let childID;
+    try {
+      childID = await wasmfsOPFSGetOrCreateDir(parent, name, true);
```

Comment:
wasmfsOPFSGetOrCreateDir already handles errors internally by return negative error codes as the ID. We should either consistently use negative IDs or we should switch entirely to using the error pointer. I should have caught this in the previous PR, but I didn't.